Jamfile

```diff
@@ -56,15 +56,52 @@ rule linking ( properties * )
 {
 	local result ;
 
+	# ssl=openssl => crypto=libcrypto
+	# this match the previous behavior
+	if <ssl>openssl in $(properties)
+		&& ! <crypto>libcrypto in $(properties)
+	{
+		EXIT "with ssl=openssl the value for crypto must be libcrypto" ;
```

Comment:
a nice feature of boost build is that properties are associated with build targets, not with invocations of bjam. what do you think of printing this message (with ECHO) and then add <build>no to result. That would simply disable the targets with invalid/unsupported combinations.

for instance, it would be possible to run: bjam ssl=openssl,none crypto=libcrypto,builtin,libgcrypt
and it would build all combinations of those features that are valid (at least that's my understanding of <build>no)